src/main/java/com/google/cloud/tools/ide/login/GoogleLoginState.java

```diff
@@ -144,17 +160,18 @@ public HttpRequestFactory createRequestFactory() {
    * @throws IOException if something goes wrong while fetching the token
    */
   public String fetchAccessToken() throws IOException {
```

Comment:
> Having to switch the active user on the login state to perform an operation, and then switch back, is not thread safe.

We will make sure everything is thread-safe. For deploy (and other operations if any), this isn't a problem, because a refresh token returned to a deploy job won't change regardless of what happens inside `GoogleLoginState`. However, now I think it's good to make `getActiveCredential()` return a copy of the current `Credential`.

For your first paragraph, not sure I understand it.